tests/test_hierarchy.cc

```diff
@@ -0,0 +1,221 @@
+    system_matrix.reinit(system_sparsity_pattern);
+
+    // Fill the system matrix
+    dealii::QGauss<dim> const quadrature(fe_degree + 1);
+    dealii::FEValues<dim> fe_values(
+        fe, quadrature,
+        dealii::update_values | dealii::update_gradients |
+            dealii::update_quadrature_points | dealii::update_JxW_values);
+    unsigned int const dofs_per_cell = fe.dofs_per_cell;
+    unsigned int const n_q_points = quadrature.size();
+    dealii::FullMatrix<double> cell_matrix(dofs_per_cell, dofs_per_cell);
+    std::vector<dealii::types::global_dof_index> local_dof_indices(
+        dofs_per_cell);
+    for (auto cell :
+         dealii::filter_iterators(dof_handler.active_cell_iterators(),
+                                  dealii::IteratorFilters::LocallyOwnedCell()))
+    {
+      cell_matrix = 0;
+      fe_values.reinit(cell);
+
+      for (unsigned int q_point = 0; q_point < n_q_points; ++q_point)
+        for (unsigned int i = 0; i < dofs_per_cell; ++i)
+          for (unsigned int j = 0; j < dofs_per_cell; ++j)
+            cell_matrix(i, j) += fe_values.shape_grad(i, q_point) *
+                                 fe_values.shape_grad(j, q_point) *
+                                 fe_values.JxW(q_point);
+
+      cell->get_dof_indices(local_dof_indices);
+      constraints.distribute_local_to_global(cell_matrix, local_dof_indices,
+                                             system_matrix);
+    }
+  }
+
+public:
+  TestMeshEvaluator(const dealii::TrilinosWrappers::SparseMatrix &matrix)
+      : _matrix(matrix)
+  {
+  }
+
+private:
+  const dealii::TrilinosWrappers::SparseMatrix &_matrix;
+};
+
+BOOST_AUTO_TEST_CASE(hierarchy_2d)
+{
+  using Vector = dealii::LinearAlgebra::distributed::Vector<double>;
+  using MeshEvaluator = mfmg::DealIIMeshEvaluator<dim, Vector>;
+  using Operator = typename MeshEvaluator::global_operator_type;
+  using Mesh = mfmg::DealIIMesh<dim>;
+
+  MPI_Comm comm = MPI_COMM_WORLD;
+
+  Source<dim> source;
+
+  const int num_refinements = 6;
+
+  Laplace<dim, Vector> laplace(comm, 1);
+  laplace.setup_system(num_refinements);
+  laplace.assemble_system(source);
+
+  auto mesh =
+      std::make_shared<Mesh>(laplace._dof_handler, laplace._constraints);
+
+  const auto &a = laplace._system_matrix;
+  Vector solution(laplace._locally_owned_dofs, comm);
+  Vector rhs(laplace._system_rhs);
+
+  const int local_size = rhs.local_size();
+
+  std::default_random_engine generator;
+  std::uniform_real_distribution<typename Vector::value_type> distribution(0.,
+                                                                           1.);
+  for (int i = 0; i < local_size; i++)
+    solution.local_element(i) = distribution(generator);
+  a.vmult(rhs, solution);
+
+  auto params = std::make_shared<boost::property_tree::ptree>();
+  params->put("eigensolver: number of eigenvectors", 1);
+  params->put("eigensolver: tolerance", 1e-14);
+  params->put<unsigned int>("agglomeration: nx", 2);
```

Comment:
I don't think that the `unsigned int` is necessary.

Reply:
Does boost allow getting the parameters using a different type?

Reply:
When you give a default value, boost can deduce the type so you don't need to give the template parameter (ofc it doesn't hurt). I know that you can put an `int` and get an `unsigned int` (the same with `float` and `double`) so I guess you just need to be able to cast the value.